src/components/FileInput/index.tsx

```diff
@@ -0,0 +1,215 @@
+
+    const dragEnterRef = React.useRef(0);
+    const [status, setStatus] = useState<string>('No file chosen');
+    const [isBeingDraggedOver, setIsBeingDraggedOver] = React.useState(false);
+
+    const isValidFile = useCallback((fileName: string, mimeType: string, acceptString?: string) => {
+        // if there is no accept string, anything is valid
+        if (!acceptString) {
+            return true;
+        }
+        const extensionMatch = /\.\w+$/.exec(fileName);
+        const mimeMatch = /^.+\//.exec(mimeType);
+
+        const fileTypeList = acceptString.split(/,\s+/);
+        return fileTypeList.some((fileType) => {
+            // check mimeType such as image/png or image/*
+            if (mimeType === fileType || (!!mimeMatch && `${mimeMatch[0]}*` === fileType)) {
+                return true;
+            }
+            return !!extensionMatch && extensionMatch[0].toLowerCase() === fileType.toLowerCase();
+        });
+    }, []);
+
+    const handleDragEnter = useCallback(() => {
+        if (dragEnterRef.current === 0) {
+            setIsBeingDraggedOver(true);
+        }
+
+        dragEnterRef.current += 1;
+    }, [setIsBeingDraggedOver]);
+
+    const handleDragLeave = useCallback(() => {
+        dragEnterRef.current -= 1;
+
+        if (dragEnterRef.current === 0) {
+            setIsBeingDraggedOver(false);
+        }
+    }, [setIsBeingDraggedOver]);
+
+    const getStatus = useCallback((files: File[]) => {
+        if (files && files.length > 1) {
+            return `${files?.length} files selected`;
+        }
+        if (files && files.length === 0) {
+            return 'Invalid files selected';
+        }
+        if (files) {
+            return files[0]?.name;
+        }
+        return 'No file chosen';
+    }, []);
+
+    const handleChange = useCallback(
+        (_: string | undefined, __: T, e?: React.FormEvent<HTMLInputElement>) => {
+            if (e) {
+                const { files: newFiles } = (e.target as HTMLInputElement);
+                if (newFiles) {
+                    const fileList = Array.from(newFiles);
+                    const validFiles = fileList.filter((f) => isValidFile(f.name, f.type, accept));
+                    const newStatus = getStatus(validFiles);
+                    setStatus(newStatus);
+                    if (onChange) {
+                        onChange(validFiles);
+                    }
+                }
+            }
+        }, [onChange, isValidFile, accept, getStatus],
+    );
+
+    const handleDrop: React.DragEventHandler<HTMLDivElement> = useCallback((e) => {
+        e.preventDefault();
+        const fileList = Array.from(e.dataTransfer.files);
+        const validFiles = fileList.filter((f) => isValidFile(f.name, f.type, accept));
+        if (!multiple && fileList.length > 1) {
+            setStatus('Multiple file selection no allowed. Please, select a single file.');
```

Comment:
```suggestion
            setStatus('Multiple file selection not allowed. Please select a single file.');
```

Reply:
Why not just take the first file and ignore the rest?

Reply:
we can select arbitrary file types when drag and dropping files, so we must check each one of them.

Reply:
I mean, we can just take the first valid file when user selects multiple files during drag and drop.

Reply:
so only a single file should be selected during drag and drop?

Reply:
Yes. I think that's the correct behavior when user tries to drag drop a lot of elements when only one is allowed